tests/ir/graph_view_test.py

```diff
@@ -0,0 +1,41 @@
+import pathlib
+import unittest
+
+import onnx
+
+from onnxscript import ir
+
+
+class GraphViewTest(unittest.TestCase):
+    def test_it_can_be_serialized_as_graph_proto(self):
+        data_path = (
+            pathlib.Path(__file__).parent.parent.parent
+            / "testdata/e2e_models/torchscript_model/torchscript_model.onnx"
+        )
+        model_proto = onnx.load(data_path)
+        model = ir.serde.deserialize_model(model_proto)
+
+        # Obtain a subset of nodes that belong to the first layer
+        nodes = [
+            node
+            for node in model.graph
+            if node.name is not None and node.name.startswith("/model/layers.0")
+        ]
+
+        inputs = set()
+        outputs = set()
+        for node in nodes:
+            inputs.update(node.inputs)
+            outputs.update(node.outputs)
+
+        graph_inputs = sorted(inputs - outputs, key=lambda val: val.name)
+        graph_outputs = sorted(outputs - inputs, key=lambda val: val.name)
+
+        graph_view = ir.GraphView(graph_inputs, graph_outputs, nodes=nodes)
+        model = ir.Model(graph_view, ir_version=8)
```

Comment:
Is cloning happening in this step?

Reply:
There will not be cloning. It will just be a reference

Reply:
What happens then if nodes inside the new model model are mutated? e.g., inputs rewired

Reply:
The graphview will change as well if the underlying nodes change, Making that clear

Reply:
Updated documentation.

> If the underlying nodes / connections are mutated, the mutation will be reflected in all views as well.

Reply:
One won't be able to insert nodes into m2. They will only be able to insert nodes to `m1.graph`.

Reply:
great, let's put that in docstring as well.

Reply:
I think it is clear in the following sense:
1. Models do not have mutation methods.
2. A model built with a graph view do not have mutation methods in its graph.

so users won't be able to modify the graph.

Reply:
Added:

>    The model created with a GraphView will have a fixed topology, and its graph
>    will remain read-only as a GraphView. No copying will be done during the
>    initialization process.